packages/plugins/plugin-inbox/src/functions/gmail-sync.ts

```diff
@@ -2,94 +2,116 @@
+  outputSchema: Schema.Struct({
+    newMessages: Schema.Number,
   }),
-
-  handler: ({ context: { space }, data: { mailboxId, userId, after, pageSize } }: any) =>
+  handler: ({
+    data: { mailboxId, userId = 'me', after = format(subDays(new Date(), 30), 'yyyy-MM-dd'), pageSize = 100 },
+  }) =>
     Effect.gen(function* () {
-      yield* Effect.log('running gmail sync', { mailboxId, userId, after, pageSize });
+      yield* Console.log('running gmail sync', { mailboxId, userId, after, pageSize });
+
+      const {
+        objects: [{ token }],
+      } = yield* DatabaseService.runQuery(Filter.type(DataType.AccessToken, { source: 'gmail.com' }));
+      yield* Console.log('found token', { token });
 
-      const { token } = yield* Effect.tryPromise({
-        try: () => space.db.query(Filter.typename('dxos.org/type/AccessToken', { source: 'gmail.com' })).first(),
-        catch: (e: any) => e,
-      });
+      // TODO(wittjosiah): Can't use `HttpClient.execute` directly because it results in CORS errors when traced.
+      //   Is this an issue on Google's side or is it a bug in `@effect/platform`?
+      const httpClient = yield* HttpClient.HttpClient;
+      const httpClientWithTracerDisabled = httpClient.pipe(HttpClient.withTracerDisabledWhen(() => true));
 
       // NOTE: Google API bundles size is v. large and caused runtime issues.
       const makeRequest = (url: string) =>
-        pipe(
-          url,
-          HttpClientRequest.get,
-          HttpClientRequest.setHeaders({ Authorization: `Bearer ${token}`, Accept: 'application/json' }),
-          HttpClient.execute,
-          Effect.flatMap((res: any) => res.json),
+        HttpClientRequest.get(url).pipe(
+          HttpClientRequest.setHeader('authorization', `Bearer ${token}`),
+          HttpClientRequest.setHeader('accept', 'application/json'),
+          httpClientWithTracerDisabled.execute,
+          Effect.flatMap((res) => res.json),
           Effect.timeout('1 second'),
           Effect.retry(Schedule.exponential(1000).pipe(Schedule.compose(Schedule.recurs(3)))),
           Effect.scoped,
         );
 
+      // TODO(wittjosiah): Add schema to parse response.
       const listMessages = (userId: string, q: string, pageSize: number, pageToken: string | null) =>
-        makeRequest(getUrl(userId, undefined, { q, pageSize, pageToken }));
-
-      const getMessage = (userId: string, messageId: string) => makeRequest(getUrl(userId, messageId));
-
-      const mailbox = yield* Effect.tryPromise({
-        try: () => space.db.query({ id: mailboxId }).first(),
-        catch: (e: any) => e,
-      });
-      const { objects } = yield* Effect.tryPromise({
-        try: () => space.queues.queryQueue(mailbox.queue.dxn),
-        catch: (e: any) => e,
-      });
-      const newMessages = yield* Ref.make([]);
-      const nextPage = yield* Ref.make(null);
+        makeRequest(getUrl(userId, undefined, { q, pageSize, pageToken })) as Effect.Effect<
```

Comment:
might be cleaner to add a generic parameter to `makeRequest`. Even better would be to use effect-schema for response

Reply:
yes, I would like to do that, I have a todo for parsing responses in there